cfgov/unprocessed/js/modules/multi-select.js

```diff
@@ -0,0 +1,494 @@
+            'data-option': option.value
+          } );
+
+          $.create( 'label', {
+            'for': option.value,
+            'innerHTML': option.text,
+            inside: li
+          } );
+
+          self.choices.appendChild( li );
+          self.selections.push( option );
+        }
+      }
+
+      self.index = -1;
+      self.skipBlur = false;
+
+      if ( self.fieldset.getAttribute( 'visibility' ) === 'visible' ) {
+        self.search.focus();
+      }
+    },
+
+    evaluate: function() {
+      var self = this;
+      var value = this.search.value;
+
+      this.resetFilter();
+
+      if ( value.length >= this.minChars
+           && this.optionData.length > 0 ) {
+        this.index = -1;
+
+        this.filtered = this.optionData.filter( function( item ) {
+          return self.filter( item.text, value );
+        } );
+
+        this.filterResults();
+      }
+    },
+
+    resetSearch: function() {
+      this.search.value = '';
+      this.resetFilter();
+    },
+
+    resetFilter: function() {
+      var self = this;
+
+      this.list.classList.remove( 'filtered', 'no-results' );
+
+      forEach( this.list.children, function( i ) {
+        self.list.children[i].classList.remove( 'filter-match' );
+      } );
+      this.filtered = this.optionData;
+    },
+
+    filterResults: function() {
+      var self = this;
+
+      this.list.classList.add( 'filtered' );
+
+      if ( this.filtered.length > 0 ) {
+        this.filtered.forEach( function( option, i ) {
+          var item = self.list.querySelector( 'li[data-option="' + option.value + '"]' );
+
+          $( item ).classList.add( 'filter-match' );
+        } );
+      } else {
+        this.noResults();
+      }
+    },
+
+    noResults: function() {
+      this.list.classList.add( 'no-results' );
+      this.list.classList.remove( 'filtered' );
+    },
+
+    bindEvents: function() {
+      var self = this;
+      var inputs = this.list.querySelectorAll( 'input' );
+
+      var things = $( '.form-l' ).querySelectorAll( '*' );
+
+      $.bind( this.search, {
+        input: this.evaluate.bind( this ),
+        focus: this.open.bind( this ),
+        blur: function( event ) {
+          if ( !self.skipBlur ) {
+            self.close();
+          }
+        },
+        mousedown: function() {
+          if ( self.fieldset.getAttribute( 'visibility' ) === 'hidden' ) {
+            self.open();
+          }
+        },
+        keydown: function( event ) {
+          var key = event.keyCode;
+
+          if ( self.fieldset.getAttribute( 'visibility' ) === 'hidden' ) {
+            self.open();
+          }
+
+          if ( key === 13 ) {
+            event.preventDefault();
+            self.next();
+          } else if ( key === 27 ) {
+            self.close();
+          } else if ( key === 38 ) {
+            event.preventDefault();
+            self.previous();
+          } else if ( key === 40 ) {
+            event.preventDefault();
```

Comment:
Probably, it'll be easier to be sure once I switch to the constants as you suggested.